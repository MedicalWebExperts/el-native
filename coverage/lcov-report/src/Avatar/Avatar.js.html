<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/Avatar/Avatar.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">src/Avatar</a> Avatar.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>12/12</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>14/14</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>12/12</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">13x</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">13x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import { oneOf, oneOfType, string, number, imgSrcPropType, bool } from 'prop-types';
import { Image } from 'react-native';
&nbsp;
import Styles from './styles';
&nbsp;
const propTypes = {
  /**
   * The prop "square" is a boolean that when present makes the Avatar squared.
   * In abscence the shape will be rounded.
   */
  square: bool,
  /**
   * The prop "size" offers 3 custom sizes to choose from. In abscence the default is medium.
   * The other sizes available are "large" and "small"
   */
  size: oneOf(['large', 'medium', 'small']),
  /**
   * The prop "borderColor" defines the color of the border.
   * This prop will not work without the "borderThickness" prop present.
   */
  borderColor: string,
  /**
   * The prop "borderThickness" determines the width of the border of the Avatar.
   * Expects a number. The default is 0.
   */
  borderThickness: number,
  /**
   * The prop "placeholder" can be used as a default image if no image is present in "source" prop.
   */
  placeholder: oneOfType([string, imgSrcPropType]),
  /**
   * The prop "source" is the main prop, and expects a string or a uri to show the image in Avatar.
   */
  source: string,
};
&nbsp;
const defaultProps = {
  square: false,
  size: 'medium',
  borderColor: '',
  borderThickness: 0,
  placeholder: null,
  source: '',
};
&nbsp;
const getSize = (size) =&gt; {
  if (size === 'large') {
    return Styles.avatarLarge;
  } else if (size === 'small') {
    return Styles.avatarSmall;
  }
  return false;
};
&nbsp;
/**
 * Avatar component is a wrapper over RN &lt;Image&gt; with custom styles
 * applied and with an interface that let users change certain props with ease.
 */
const Avatar = props =&gt; (
  &lt;Image
    source={props.source ? { uri: props.source } : props.placeholder}
    style={[
      Styles.avatar,
      !!props.borderThickness &amp;&amp; { borderWidth: props.borderThickness },
      !!props.borderColor &amp;&amp; { borderColor: props.borderColor },
      !!props.size &amp;&amp; getSize(props.size),
      !!props.square &amp;&amp; Styles.avatarSquare,
    ]}
  /&gt;
);
&nbsp;
Avatar.propTypes = propTypes;
Avatar.defaultProps = defaultProps;
&nbsp;
export default Avatar;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Jan 17 2018 12:58:56 GMT-0300 (-03)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
