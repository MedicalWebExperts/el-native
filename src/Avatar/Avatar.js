import React from 'react';
import { oneOf, oneOfType, string, number, imgSrcPropType, bool } from 'prop-types';
import { Image } from 'react-native';

import Styles from './styles';

const propTypes = {
  /**
   * Description of prop "square".
   */
  square: bool,
  /**
   * Description of prop "size".
   */
  size: oneOf(['large', 'medium', 'small']),
  /**
   * Description of prop "borderColor".
   */
  borderColor: string,
  /**
   * Description of prop "borderThickness".
   */
  borderThickness: number,
  /**
   * Description of prop "placeholder".
   */
  placeholder: oneOfType([string, imgSrcPropType]),
  /**
   * Description of prop "source".
   */
  source: string,
};

const defaultProps = {
  square: false,
  size: 'medium',
  borderColor: '',
  borderThickness: 0,
  placeholder: null,
  source: '',
};

const getSize = (size) => {
  if (size === 'large') {
    return Styles.avatarLarge;
  } else if (size === 'small') {
    return Styles.avatarSmall;
  }
  return false;
};

/**
 * Avatar component description.
 */
const Avatar = props => (
  <Image
    source={props.source ? { uri: props.source } : props.placeholder}
    style={[
      Styles.avatar,
      !!props.borderThickness && { borderWidth: props.borderThickness },
      !!props.borderColor && { borderColor: props.borderColor },
      !!props.size && getSize(props.size),
      !!props.square && Styles.avatarSquare,
    ]}
  />
);

Avatar.propTypes = propTypes;
Avatar.defaultProps = defaultProps;

export default Avatar;
