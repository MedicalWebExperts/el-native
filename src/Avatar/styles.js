export default {
  avatar: {
    width: 60,
    height: 60,
    borderRadius: 30,
  },
};
