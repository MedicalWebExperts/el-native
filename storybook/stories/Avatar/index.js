import React from 'react';
import { storiesOf } from '@storybook/react-native';

import CenterView from '../CenterView';
import { Avatar } from '../../../src';
import placeholderImage from '../../../src/Avatar/placeholderImage.png';
import colors from '../../../src/styles/colors';

const imageUrl = 'https://wallscover.com/images/gunther-wallpaper-2.jpg';

storiesOf('Avatar', module)
  .addDecorator(getStory => <CenterView>{getStory()}</CenterView>)
  .add('Large', () => <Avatar source={imageUrl} size="large" />)
  .add('Large (placeholder example)', () => <Avatar placeholder={placeholderImage} size="large" />)
  .add('Medium (default)', () => <Avatar source={imageUrl} />)
  .add('Small', () => <Avatar source={imageUrl} size="small" />)
  .add('Large Bordered', () => (
    <Avatar source={imageUrl} size="large" borderColor={colors.blueColor} borderThickness={2} />
  ))
  .add('Medium bordered', () => (
    <Avatar source={imageUrl} size="medium" borderColor={colors.blueColor} borderThickness={2} />
  ))
  .add('Small Bordered', () => (
    <Avatar source={imageUrl} size="small" borderColor={colors.blueColor} borderThickness={2} />
  ))
  .add('Large Square', () => <Avatar source={imageUrl} size="large" square />)
  .add('Medium Square', () => <Avatar source={imageUrl} size="medium" square />)
  .add('Small Square', () => <Avatar source={imageUrl} size="small" square />)
  .add('Large Square bordered', () => (
    <Avatar
      source={imageUrl}
      square
      size="large"
      borderColor={colors.blueColor}
      borderThickness={2}
    />
  ))
  .add('Medium Square bordered', () => (
    <Avatar
      source={imageUrl}
      square
      size="medium"
      borderColor={colors.blueColor}
      borderThickness={2}
    />
  ))
  .add('Small Square bordered', () => (
    <Avatar
      source={imageUrl}
      square
      size="small"
      borderColor={colors.blueColor}
      borderThickness={2}
    />
  ));
