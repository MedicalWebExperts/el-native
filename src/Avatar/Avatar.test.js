import React from 'react';
import renderer from 'react-test-renderer';

import Avatar from './Avatar';

import placeholderImage from './placeholderImage.png';

const imageUrl = 'https://wallscover.com/images/gunther-wallpaper-2.jpg';

const props = {
  square: false,
  size: 'medium',
  borderColor: '',
  borderThickness: 0,
  placeholder: null,
  source: '',
};

describe('Avatar Snapshots', () => {
  it('should render an Avatar Large with placeholder', () => {
    const component = renderer.create(
      <Avatar {...props} size="large" placeholder={placeholderImage} />,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
  it('should render an Avatar Large', () => {
    const component = renderer.create(<Avatar {...props} source={imageUrl} size="large" />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
  it('should render an Avatar Medium', () => {
    const component = renderer.create(<Avatar {...props} source={imageUrl} />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
  it('should render an Avatar Small', () => {
    const component = renderer.create(<Avatar {...props} source={imageUrl} size="small" />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
  it('should render an Avatar Large bordered', () => {
    const component = renderer.create(
      <Avatar {...props} source={imageUrl} size="large" borderColor={'blue'} borderThickness={2} />,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
  it('should render an Avatar Medium bordered', () => {
    const component = renderer.create(
      <Avatar
        {...props}
        source={imageUrl}
        size="medium"
        borderColor={'blue'}
        borderThickness={2}
      />,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
  it('should render an Avatar Small bordered', () => {
    const component = renderer.create(
      <Avatar {...props} source={imageUrl} size="small" borderColor={'blue'} borderThickness={2} />,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
  it('should render an Avatar Square Large', () => {
    const component = renderer.create(<Avatar {...props} source={imageUrl} square size="large" />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
  it('should render an Avatar Square Medium', () => {
    const component = renderer.create(<Avatar {...props} source={imageUrl} square size="medium" />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
  it('should render an Avatar Square Small', () => {
    const component = renderer.create(<Avatar {...props} source={imageUrl} square size="small" />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
  it('should render an Avatar Square Large bordered', () => {
    const component = renderer.create(
      <Avatar
        {...props}
        source={imageUrl}
        square
        size="large"
        borderColor={'blue'}
        borderThickness={2}
      />,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
  it('should render an Avatar Square Medium bordered', () => {
    const component = renderer.create(
      <Avatar
        {...props}
        source={imageUrl}
        square
        size="medium"
        borderColor={'blue'}
        borderThickness={2}
      />,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
  it('should render an Avatar Square Small bordered', () => {
    const component = renderer.create(
      <Avatar
        {...props}
        source={imageUrl}
        square
        size="small"
        borderColor={'blue'}
        borderThickness={2}
      />,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
