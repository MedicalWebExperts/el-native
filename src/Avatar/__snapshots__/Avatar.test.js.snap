// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Snapshots should render an Avatar Large 1`] = `
<Image
  source={
    Object {
      "uri": "https://wallscover.com/images/gunther-wallpaper-2.jpg",
    }
  }
  style={
    Array [
      Object {
        "borderRadius": 30,
        "height": 60,
        "width": 60,
      },
      false,
      false,
      Object {
        "borderRadius": 45,
        "height": 90,
        "width": 90,
      },
      false,
    ]
  }
/>
`;

exports[`Avatar Snapshots should render an Avatar Large bordered 1`] = `
<Image
  source={
    Object {
      "uri": "https://wallscover.com/images/gunther-wallpaper-2.jpg",
    }
  }
  style={
    Array [
      Object {
        "borderRadius": 30,
        "height": 60,
        "width": 60,
      },
      Object {
        "borderWidth": 2,
      },
      Object {
        "borderColor": "#2196F3",
      },
      Object {
        "borderRadius": 45,
        "height": 90,
        "width": 90,
      },
      false,
    ]
  }
/>
`;

exports[`Avatar Snapshots should render an Avatar Large with placeholder 1`] = `
<Image
  source={
    Object {
      "testUri": "../../../src/Avatar/placeholderImage.png",
    }
  }
  style={
    Array [
      Object {
        "borderRadius": 30,
        "height": 60,
        "width": 60,
      },
      false,
      false,
      Object {
        "borderRadius": 45,
        "height": 90,
        "width": 90,
      },
      false,
    ]
  }
/>
`;

exports[`Avatar Snapshots should render an Avatar Medium 1`] = `
<Image
  source={
    Object {
      "uri": "https://wallscover.com/images/gunther-wallpaper-2.jpg",
    }
  }
  style={
    Array [
      Object {
        "borderRadius": 30,
        "height": 60,
        "width": 60,
      },
      false,
      false,
      false,
      false,
    ]
  }
/>
`;

exports[`Avatar Snapshots should render an Avatar Medium bordered 1`] = `
<Image
  source={
    Object {
      "uri": "https://wallscover.com/images/gunther-wallpaper-2.jpg",
    }
  }
  style={
    Array [
      Object {
        "borderRadius": 30,
        "height": 60,
        "width": 60,
      },
      Object {
        "borderWidth": 2,
      },
      Object {
        "borderColor": "#2196F3",
      },
      false,
      false,
    ]
  }
/>
`;

exports[`Avatar Snapshots should render an Avatar Small 1`] = `
<Image
  source={
    Object {
      "uri": "https://wallscover.com/images/gunther-wallpaper-2.jpg",
    }
  }
  style={
    Array [
      Object {
        "borderRadius": 30,
        "height": 60,
        "width": 60,
      },
      false,
      false,
      Object {
        "borderRadius": 15,
        "height": 30,
        "width": 30,
      },
      false,
    ]
  }
/>
`;

exports[`Avatar Snapshots should render an Avatar Small bordered 1`] = `
<Image
  source={
    Object {
      "uri": "https://wallscover.com/images/gunther-wallpaper-2.jpg",
    }
  }
  style={
    Array [
      Object {
        "borderRadius": 30,
        "height": 60,
        "width": 60,
      },
      Object {
        "borderWidth": 2,
      },
      Object {
        "borderColor": "#2196F3",
      },
      Object {
        "borderRadius": 15,
        "height": 30,
        "width": 30,
      },
      false,
    ]
  }
/>
`;

exports[`Avatar Snapshots should render an Avatar Square Large 1`] = `
<Image
  source={
    Object {
      "uri": "https://wallscover.com/images/gunther-wallpaper-2.jpg",
    }
  }
  style={
    Array [
      Object {
        "borderRadius": 30,
        "height": 60,
        "width": 60,
      },
      false,
      false,
      Object {
        "borderRadius": 45,
        "height": 90,
        "width": 90,
      },
      Object {
        "borderRadius": 0,
      },
    ]
  }
/>
`;

exports[`Avatar Snapshots should render an Avatar Square Large bordered 1`] = `
<Image
  source={
    Object {
      "uri": "https://wallscover.com/images/gunther-wallpaper-2.jpg",
    }
  }
  style={
    Array [
      Object {
        "borderRadius": 30,
        "height": 60,
        "width": 60,
      },
      Object {
        "borderWidth": 2,
      },
      Object {
        "borderColor": "#2196F3",
      },
      Object {
        "borderRadius": 45,
        "height": 90,
        "width": 90,
      },
      Object {
        "borderRadius": 0,
      },
    ]
  }
/>
`;

exports[`Avatar Snapshots should render an Avatar Square Medium 1`] = `
<Image
  source={
    Object {
      "uri": "https://wallscover.com/images/gunther-wallpaper-2.jpg",
    }
  }
  style={
    Array [
      Object {
        "borderRadius": 30,
        "height": 60,
        "width": 60,
      },
      false,
      false,
      false,
      Object {
        "borderRadius": 0,
      },
    ]
  }
/>
`;

exports[`Avatar Snapshots should render an Avatar Square Medium bordered 1`] = `
<Image
  source={
    Object {
      "uri": "https://wallscover.com/images/gunther-wallpaper-2.jpg",
    }
  }
  style={
    Array [
      Object {
        "borderRadius": 30,
        "height": 60,
        "width": 60,
      },
      Object {
        "borderWidth": 2,
      },
      Object {
        "borderColor": "#2196F3",
      },
      false,
      Object {
        "borderRadius": 0,
      },
    ]
  }
/>
`;

exports[`Avatar Snapshots should render an Avatar Square Small 1`] = `
<Image
  source={
    Object {
      "uri": "https://wallscover.com/images/gunther-wallpaper-2.jpg",
    }
  }
  style={
    Array [
      Object {
        "borderRadius": 30,
        "height": 60,
        "width": 60,
      },
      false,
      false,
      Object {
        "borderRadius": 15,
        "height": 30,
        "width": 30,
      },
      Object {
        "borderRadius": 0,
      },
    ]
  }
/>
`;

exports[`Avatar Snapshots should render an Avatar Square Small bordered 1`] = `
<Image
  source={
    Object {
      "uri": "https://wallscover.com/images/gunther-wallpaper-2.jpg",
    }
  }
  style={
    Array [
      Object {
        "borderRadius": 30,
        "height": 60,
        "width": 60,
      },
      Object {
        "borderWidth": 2,
      },
      Object {
        "borderColor": "#2196F3",
      },
      Object {
        "borderRadius": 15,
        "height": 30,
        "width": 30,
      },
      Object {
        "borderRadius": 0,
      },
    ]
  }
/>
`;
